lib/src/interceptors/http_interceptor.dart

```diff
@@ -1 +1,64 @@
-// TODO: HttpInterceptor
+import 'dart:io';
+import 'package:dio/dio.dart';
+import 'package:http_mock_adapter/src/handlers/request_handler.dart';
+import 'package:http_mock_adapter/src/history.dart';
+import 'package:http_mock_adapter/src/request.dart';
+import 'package:http_mock_adapter/src/adapter_interface.dart';
+
+/// [DioInterceptor] is a class for mocking the [Dio] requests with interceptors.
+///
+/// This means you can mock any request of [Dio] by adding the
+/// instance of [DioInterceptor] inside the original [Dio]'s interceptors' list.
+///
+/// Usage:
+/// ```dart
+/// // Create Dio instance
+/// Dio dio = Dio()
+/// // Create instance of our([DioInterceptor]) Interceptor
+/// DioInterceptor interceptor = DioInterceptor()
+/// // Adding routes and their mocked responses as chains
+/// interceptor
+/// .onGet("/route-1")
+/// .reply("response for route 1")
+/// .onPost("/route-2")
+/// .reply("response for route 2")
+/// .onPatch("/route-3")
+/// .reply("response for route 3")
+/// // adding intercetor inside the [Dio]'s interceptors
+/// dio.interceptors.add(interceptor);
+/// ```
+/// If you now make request like this `dio.get("/route-1");`
+/// Your response will be `Response(data:"response for route 1",........)`
+
+class DioInterceptor extends Interceptor
```

Comment:
`MainInterceptor` should be `DioInterceptor` in this file.

I advise to use an IDE and rename everything all at once.

Reply:
Naming file `dio_interceptor` would be preferable since the latest changes, I think.